File: 백준/Silver/1926. 그림/그림.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**7)
directions = [
    [-1, 0],
    [1, 0],
    [0, -1],
    [0, 1],
]

def dfs(graph, x, y):
    count = 1
    graph[x][y] = 2

    for (dx, dy) in directions:
        nx = x + dx
        ny = y + dy
        if 0 <= nx < len(graph) and 0 <= ny < len(graph[0]) and graph[nx][ny] == 1:
            count += dfs(graph, nx, ny)

    return count


n, m = map(int, input().split())  # 행, 열
graph = [list(map(int, input().split())) for _ in range(n)]

ret = []
for i in range(n):
    for j in range(m):
        if graph[i][j] == 1:
            ret.append(dfs(graph, i, j))

if ret:
    print(len(ret))
    print(max(ret))
else:
    print(0)
    print(0)

File: 백준/Silver/1926. 그림/그림.kt
package org.example

import java.util.StringTokenizer
import java.util.PriorityQueue

// 상, 하, 좌, 우
private val dx = intArrayOf(-1, 1, 0, 0)
private val dy = intArrayOf(0, 0, -1, 1)

fun main() {
    val br = System.`in`.bufferedReader()
    val (n, m) = br.readLine().split(" ").map { it.toInt() }
    val graph = Array(n) { IntArray(m) }
    for (i in 0 until n) {
        val st = StringTokenizer(br.readLine())
        for (j in 0 until m) {
            graph[i][j] = st.nextToken().toInt()
        }
    }
    val pq = PriorityQueue<Int>(compareByDescending { it })
    for (i in 0 until n) {
        for (j in 0 until m) {
            if (graph[i][j] == 1) {
                pq.add(dfs(graph, i, j))
            }
        }
    }
    if(pq.isEmpty()){
        println("0\n0")
    }else{
        println(pq.size)
        println(pq.poll())
    }
    br.close()
}

fun dfs(graph: Array<IntArray>, x: Int, y: Int): Int {
    var count = 1
    graph[x][y] = 0
    for (i in 0 until 4) {
        val nx = x + dx[i]
        val ny = y + dy[i]
        if (nx in graph.indices && ny in graph[0].indices && graph[nx][ny] == 1) {
            count += dfs(graph, nx, ny)
        }
    }
    return count
}
